packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -658,11 +659,35 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
 
   if (eventType.schedulingType === SchedulingType.COLLECTIVE) {
     evt.team = {
-      members: users.map((user) => user.name || user.username || "Nameless"),
+      members: users.map((user) => {
+        return { name: user.name || user.username || "Nameless", email: user.email, id: user.id };
+      }),
       name: eventType.team?.name || "Nameless",
     }; // used for invitee emails
   }
 
+  if (eventType.schedulingType === SchedulingType.ROUND_ROBIN) {
+    const teamMembers = [] as { id: number; email: string; name: string }[];
+    const filteredAttendees = evt.attendees;
+    evt.attendees.forEach((attendee) => {
+      const bookingAttendee = evt.attendees.find(
+        (bookingAttendee) => bookingAttendee.email === attendee.email
+      );
+      if (bookingAttendee) {
+        const member = users.find((teamMember) => teamMember.email === bookingAttendee.email);
+        if (member) {
+          teamMembers.push({ id: member.id, email: member.email, name: member.name || "Nameless" });
+          filteredAttendees.filter((filteredAttendee) => filteredAttendee.email !== member.email);
+        }
+      }
+    });
+    evt.attendees = filteredAttendees;
+    evt.team = {
+      members: teamMembers,
+      name: eventType.team?.name || "Nameless",
+    };
+  }
```

Comment:
When creating a new round robin booking, team members are added as `attendees`. They should be moved over to the `team.members` object.